src/main/java/bio/terra/workspace/service/workspace/flight/GenerateResourceIdStep.java

```diff
@@ -0,0 +1,34 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.RBS_RESOURCE_ID;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+
+/**
+ * Generates Resource Id for communication with RBS and put it in working map.
+ *
+ * <p>This is important to do as a separate step from actually calling RBS so that if there is a
+ * failure we can hand RBS the same resource ID or know which one to delete.
+ */
+public class GenerateResourceIdStep implements Step {
+  public GenerateResourceIdStep() {}
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) {
+    flightContext.getWorkingMap().put(RBS_RESOURCE_ID, randomResourceId());
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext flightContext) {
+    return StepResult.getStepResultSuccess();
+  }
+
+  /** Generate a pseudo-random resource id */
+  public static String randomResourceId() {
+    return "wm-" + Long.valueOf(UUID.randomUUID().getMostSignificantBits()).toString();
```

Comment:
I'm not familiar with RBS so this is mostly for my knowledge - is this an established format for resource ID requests?

Reply:
@yonghaoy I am actually not sure, this just seemed reasonable. Thoughts?

Reply:
any unique random string works! I am not famillar with `getMostSignificantBits()` method. Is that large enough to avoid collision? If yes, this method LG!